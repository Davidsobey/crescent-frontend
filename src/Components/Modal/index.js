import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Typography } from 'material-ui';
import ModalPaper from './styles';

class CustomModal extends React.Component {
  constructor(props) {
    super(props);
    this.handleOpen = this.handleOpen.bind(this);
  }

  state = {
    open: false,
  };

  componentDidMount() {
    this.props.onRef(this);
  }
  componentWillUnmount() {
    this.props.onRef(null);
  }

  handleOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  loadRows = (obj, arr) => {
    if (obj) {
      Object.keys(obj).forEach((prop) => {
        if (obj[prop] && !Array.isArray(obj[prop])) {
          arr.push(obj[prop]);
        }
      });
    }
    return arr;
  };

  render() {
    const style = {
      justifyContent: 'center',
      alignItems: 'center',
    };
    const { del, obj } = this.props;
    return (
      <Modal
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={this.state.open}
        onClose={this.handleClose}
        style={style}
      >
        <ModalPaper>
          <Typography variant="title" id="modal-title">
            {`Are you sure you want to delete this ${del}?`}
          </Typography>
          {this.loadRows(obj, []).map(returnObj => (
            <Typography
              key={returnObj}
              variant="subheading"
              id="simple-modal-description"
            >
              {returnObj}
            </Typography>
          ))}
        </ModalPaper>
      </Modal>
    );
  }
}

CustomModal.propTypes = {
  del: PropTypes.string,
  obj: PropTypes.object,
  onRef: PropTypes.func,
};

export default CustomModal;
