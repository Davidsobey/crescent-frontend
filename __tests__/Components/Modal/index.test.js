import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import Modal from '../../../src/Components/Modal/index.js';


const mockStore = configureStore();
const initialState = {};
const store = mockStore(initialState);

describe('<Modal />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const mockOnClick = jest.fn();
      const mockOnRef = jest.fn();
      const wrapper = shallow(<Modal store={store} onRef={mockOnRef} onClick={mockOnClick} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});