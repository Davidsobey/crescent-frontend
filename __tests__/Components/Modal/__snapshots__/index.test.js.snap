// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> render() renders the component 1`] = `
<Modal
  BackdropComponent={[Function]}
  aria-describedby="simple-modal-description"
  aria-labelledby="simple-modal-title"
  classes={
    Object {
      "hidden": "MuiModal-hidden-2",
      "root": "MuiModal-root-1",
    }
  }
  disableAutoFocus={false}
  disableBackdropClick={false}
  disableEnforceFocus={false}
  disableEscapeKeyDown={false}
  disableRestoreFocus={false}
  hideBackdrop={false}
  keepMounted={false}
  manager={
    ModalManager {
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "modals": Array [],
    }
  }
  onClose={[Function]}
  open={false}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <styled.div>
    <WithStyles(Typography)
      id="modal-title"
      variant="title"
    >
      Are you sure you want to delete this:
    </WithStyles(Typography)>
    <br />
    <br />
    <div
      className="justify-content flex-container"
    >
      <WithTheme(ButtonWithTheme)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithTheme(ButtonWithTheme)>
      <WithTheme(ButtonWithTheme)
        color="primary"
        onClick={[Function]}
        variant="raised"
      >
        Confirm
      </WithTheme(ButtonWithTheme)>
    </div>
  </styled.div>
</Modal>
`;
